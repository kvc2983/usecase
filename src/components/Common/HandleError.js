import React from 'react';

const HandleError = ({ error }) => {
    return (
        <div>
            <p>OOPS! Something went Wrong:</p>
            <p style={{ color: 'red' }}>{error}</p>
            <p>Please try again after sometime</p>
        </div>
    )
}

export default HandleError; 