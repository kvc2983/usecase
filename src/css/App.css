.App{
    padding: 0px 20px 0px 0px;
}

/*
Splash
*/
.splash{}
.splash img{
    height: 100px;
}

.app-cotent {
    margin-top: 40px;
    height: 600px;
    width: 1500px;
}

/*
VIDEO PLAYER
*/
.player{
    width: 100%;
    height: 55%;
    background: #000;
}
.player video{
    width: 100%;
    height: 100%;
    object-fit: cover;
}
.player .ctrls{
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 90px;
    background: rgba(62, 62, 62, 0.54);
    z-index: 9;
    padding: 0px 15px;
}
.player .ctrls .pp{
    border: 0px;
    width: 50px;
    height: 50px;
    border-radius: 10px;
    background: #ea5252;
    color: #fff;    
}
.player .ctrls .timer{
    flex: 1;
    margin: 0px;
    text-align: center;
}
.player .ctrls .slider{
    flex: 3;
}
.player .ctrls .slider .prog{
    width: 100%;
    height: 10px;
    background: #ffffff;
    border-radius: 10px;
}
.player .ctrls .slider .bar{
    width: 40%;
    height: 10px;
    background: #ea5252;
    border-radius: 10px;
}
.player .ctrls .slider .bar .knob{
    width: 20px;
    height: 20px;
    background: #fff;
    border-radius: 100%;
    top: 50%;
    right: 0px;
    transform: translateY(-50%);
    box-shadow: 0px 0px 5px #000;
}
.player .ctrls .vol{
    border: 0px;
    width: 50px;
    height: 50px;
    border-radius: 10px;
    background: rgba(234, 82, 82, 0);
    color: #fff;
    margin: 0px 10px;
}
.player .ctrls .fs{
    border: 0px;
    width: 50px;
    height: 50px;
    border-radius: 10px;
    background: rgba(234, 82, 82, 0);
    color: #fff;
    margin: 0px 10px;
}


.allpage {
    max-width: 555px;
    padding: 10px;
}

.fontenroll{
    padding-bottom: 10px;
}

.tutor {
    padding-top: 10px;
}

.enroll {
    height: 27px;
    margin-top: 315px;
}
.remove {
    height: 27px;
    margin-top: 36px;
}

.video {
    height: 370px;
    width: 470px;
    margin-right: 10px;
}