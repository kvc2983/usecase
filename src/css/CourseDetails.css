/*
COURSE PREVIEW
*/
.course-page{}
.course-info{flex: 1;}
.course-page .tutor{}
.course-page .tutor .pic{
    width: 40px;
    height: 40px;
    border-radius: 10px;
    background: #eee;
    margin-right: 10px;
}
.course-page .tutor .pic img{
    width: 100%;
    height: 100%;
    border-radius: 10px;
}
.course-page .tutor .meta{}
.course-page .tutor .meta .name{}
.course-page .tutor .meta .uname{}
.course-page .course-meta{
    padding: 40px 0px;
}
.course-page .course-meta .title{
    font-weight: 600;
}
.course-page .course-meta .about{}

.course-videos{
    flex-direction: column;
    align-items: flex-start;
    margin: 20px 0px;
}
.course-videos a{
    margin: 10px 0px;
}
.course-videos a .id{
    background: #ea5252;
    width: 50px;
    height: 50px;
    border-radius: 14px;
    padding: 14px 20px;
    box-sizing: border-box;
    margin-right: 10px;
}
.course-videos a .meta{}
.course-videos a .meta .lbl{}
.course-videos a .meta .dur{}
.course-videos a:hover{}

.course-preview{
    flex: 2;
    margin-left: 20px;
}

.video-align{
    margin-right: 12px;
}

.extra-block{}
.chat{
    flex: 1;
    margin-right: 50px;
}
.chat .messages{
    flex-direction: column;
    text-align: left;
    justify-content: flex-start;
    padding: 30px 0px;
}
.chat .messages .bubble{
    align-self: flex-start;
    margin: 7px 0px;
    width: 100%;
}
.chat .messages .bubble .txt{
    background: #d6d6d6;
    line-height: 1;
    padding: 14px;
    border-radius: 20px 20px 20px 3px;
}
.chat .messages .bubble-mine{
    text-align: right;
}
.chat .messages .bubble-mine .txt{
    border-radius: 20px 20px 3px 20px !important;
    background: #272727;
    color: #fff;
}

.now-watching{
    flex: 1;
}
.now-watching .tooltip{
    background: #ea5252;
    top: 50px;
    right: 0px;
    padding: 8px 15px;
    border-radius: 10px;
}
.now-watching .you-list{margin-top :25px;}
.now-watching .you{
    margin-bottom: 30px;
}
.now-watching .you .pic{
    width: 40px;
    height: 40px;
    margin-right: 12px;
    background: #ddd;
    border-radius: 100%;
}
.now-watching .you .pic img{
    width: 100%;
    height: 100%;
    border-radius: 100%;
}
.now-watching .you .meta{}
.now-watching .you .meta .name{}
.now-watching .you .meta .uname{}