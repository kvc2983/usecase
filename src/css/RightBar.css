/*
RIGHTBAR
*/
.rightbar{
    flex: 1;
    padding: 20px 0px;
    max-width: 275px;
}

.rightbar .search-box{
    margin: 10px 0px;
}
.rightbar .search-box .qry{
    flex: 1;
    border: 0px;
    padding: 15px;
    background: #f1f1f1;
    margin-right: 10px;
    border-radius: 5px;
}
.rightbar .search-box .qry:focus{
    background: #e0e0e0;
}
.rightbar .search-box .go{
    border: 0px;
    background: #ea5252;
    padding: 0px 20px;
    border-radius: 5px;
}
.rightbar .search-box .go:hover{
    background: #d24444;
}
.rightbar .courses{
    flex-direction: column;
}
.rightbar .courses .cols{
    flex-direction: row;
}
.rightbar .course-a .block{
    width: 100%;
    height: 250px;
    margin-bottom: 25px;
}
.rightbar .courses .cols .block{
    height: 290px;
}

.rightbar .special-offer{
    background: #e5fcfb;
    height: 185px;
    margin-top: 14px;
    border-radius: 20px;
    overflow: hidden;
}
.rightbar .special-offer .meta{
    padding: 30px;
    flex: 1;
    z-index: 2;
}
.rightbar .special-offer .meta .h1{color: #ea5252;}
.rightbar .special-offer .meta .h2{
    margin-top: 6px;
    color: #06a9a2;
}
.rightbar .special-offer .meta button{
    background: #fff;
    border: 0px;
    border-radius: 10px;
    padding: 14px 20px;
    margin-top: 20px;
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
}
.rightbar .special-offer .meta button:hover{
    background: #ea5252;
    color: white;
}
.rightbar .special-offer .vector{
    padding-top: 30px;
    width: 150px;
    z-index: 2;
}
.rightbar .special-offer .vector img{}
.rightbar .special-offer .ring{
    width: 100px;
    height: 100px;
    background: #90e6e3;
    border-radius: 100%;
    right: -90px;
    top: 50%;
    transform: translateY(-50%);
    border: 40px rgba(255, 255, 255, 0.45) solid;
}