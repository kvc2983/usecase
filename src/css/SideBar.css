/*
SIDEBAR
*/
.sidebar{
    width: 275px;
}

.sidebar .logo{
    padding: 20px 25px;
}
.sidebar .logo img{
    height: 40px;
}

.sidebar .nav{
    padding: 50px 35px;
}
.sidebar .nav li{}
.sidebar .nav li a{
    padding: 10px 20px;
    margin-bottom: 6px;
}
.sidebar .nav li a .ico{
    margin-right: 10px;
}
.sidebar .nav li a .lbl{}

.sidebar .nav li a.active,
.sidebar .nav li a.on{
    border-left: 2px #ea5252 solid;
    color: #ea5252;
    line-height: 1;
}

.sidebar .updated-course{
    padding: 0px 35px;
    margin: 20px 0px;
}
.sidebar .updated-course .ico{
    background: #ea5252;
    padding: 15px;
    border-radius: 22px;
    margin-right: 10px;
}
.sidebar .updated-course .lbl{}
.sidebar .updated-course .lbl .author{}

.course-stats, .sidebar .stats{
    padding: 50px 50px;
}
.course-stats .stats-box, .sidebar .stats .stats-box{
    flex-direction: column;
    width: 110px;
}
.course-stats .stats-box .ico, .sidebar .stats .stats-box .ico{
    margin-bottom: 15px;
}
.sidebar .stats .stats-box .val{}
.sidebar .stats .stats-box .lbl{}
.course-stats .ico-points, .sidebar .stats .stats-box .ico-points{color: #1aab53;}
.course-stats .ico-battery, .sidebar .stats .stats-box .ico-battery{color: #ea5252;}

.sidebar .me{
    padding: 20px 40px;
}
.sidebar .me .photo{
    width: 50px;
    height: 50px;
    border-radius: 25px;
    overflow: hidden;
    margin-right: 10px;
    background: #eee;
}
.sidebar .me .photo img{
    width: 100%;
    height: 100%;
    border-radius: 50%;
}
.sidebar .me .lbl{}

.sidebar .me .link{}
.sidebar .me .link .ico{
    width: 50px;
    height: 50px;
    background: #ea5252;
    border-radius: 100%;    
    margin-right: 10px;
}
.sidebar .me .link .ico:before{
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}
.sidebar .me .link .lbl{}

.app-content{
    flex: 2;
    padding: 20px 40px 20px 20px;
}