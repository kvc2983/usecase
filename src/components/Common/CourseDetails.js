import React, {useEffect, useState} from "react";
import PropTypes from 'prop-types';
import HandleError from '../Common/HandleError'

function CourseDetails(props){

    const [course, setCourse] = useState(null);
    const [error, setError] = useState(null);
    const [enroll, setEnroll] = useState(true);
    const [isEnrolled, setIsEnrolled] = useState(true);


    //To get course Id for which details should be shown
    const courseID = props.match.params.courseid;

    //Fetching all Details of particular course
    useEffect(()=>{
        fetch('http://localhost:3000/allcourses/'+courseID)
        .then(res => {
            if(!res.ok) {
                throw Error("Endpoint for CourseDetails is failed to fetch data");
            }
            return res.json();
        })
        .then(data => {
            setCourse(data);
            setError(null);
        })
        .catch(err => {
            console.log(err.message);
            setError(err.message);
        });
    }, [courseID])

    //Check if user has allready Enrolled for this course
    useEffect(()=>{
        fetch('http://localhost:3000/mycourses/'+courseID)
        .then(res => {
            if(!res.ok) {
                throw Error("Endpoint CourseDetails is failed to fetch data");
            }
            return res.json();
        })
        .then(data => {
            data && setIsEnrolled(false);
            setError(null);
        })
        .catch(err => {
            console.log(err.message);
            setError(err.message);
        });
    }, [])

    //On click of enroll add course to my courses
    const handleEnroll = (id) => {
        fetch('http://localhost:3000/allcourses/' + id)
        .then(res => {
            return res.json();
        })
        .then(data => {
            fetch('http://localhost:3000/mycourses', {
                method: 'POST',
                body: JSON.stringify(data),
                headers: { 'Content-Type': 'application/json'}
            });
            setEnroll(false);
        })
        .catch(err => {
            console.log(err.message);
            setError(err.message);
        });
    }

    var courseVideos = [];
   if(course) {
        for(let i = 0; i < course.videos.length; i++){
            courseVideos.push(
                <a href="/#" key={"course-video-" + i} className="noul aic rel flex">
                    <div className="id s18 fontn cfff">{course.videos[i].ID}</div>
                    <div className="meta rel">
                        <h1 className="s15 lbl fontb c333">{course.videos[i].title}</h1>
                        <h1 className="s13 dur fontn c777">{course.videos[i].duration}</h1>
                    </div>
                </a>            
            );
        }
    }
    try {
        return (
            <div className="course-page rel flex">
            { error && !course ? <HandleError error={error} /> 
            : course &&
            <div className="course-info rel">

                    <div className="tutor rel aic flex">
                        <div className="pic">
                            <img src={course.tutor.dp} className="bl" alt="CourseImg" />
                        </div>
                        <div className="meta rel">
                            <h2 className="s15 name fontb c333">{course.tutor.name}</h2>
                            <h2 className="s13 uname fontn c777">Course Tutor</h2>
                        </div>
                    </div>

                    <div className="course-meta">
                        <h2 className="s24 title fontb c333">{course.title}</h2>
                        <p className="s18 about fontn c777" dangerouslySetInnerHTML={{ __html: course.about}} />
                        <div className="section section-b rel">
                            <h2 className="title s24 fontb">Course <span  className="fontn">Achievements</span></h2>
                            <div className="course-stats aic flex">
                                <div className="stats-box flex">
                                    <div className="ico ico-points s24 icon-shield" />
                                    <h2 className="val s15 c333 fontb">{course.points}</h2>
                                    <h2 className="lbl s13 c777">points</h2>
                                </div>
                                <div className="stats-box flex">
                                    <div className="ico ico-battery s24 icon-battery-90" />
                                    <h2 className="val s15 c333 fontb">{course.percent}%</h2>
                                    <h2 className="lbl s13 c777">complete</h2>
                                </div>
                                <div className="stats-box flex">
                                    <div className="ico ico-battery s24 icon-battery-90" />
                                    <h2 className="val s15 c333 fontb">{course.level}</h2>
                                    <h2 className="lbl s13 c777">level</h2>
                                </div>
                            </div>
                        </div>        
                        {
                            enroll && isEnrolled ?
                            <input type="button" className="lbl s20 enroll-course" onClick={(e) => (e.preventDefault(), handleEnroll(courseID)) } value="Enroll Now" />
                            :
                            <p className="s18 about fontn c777" dangerouslySetInnerHTML={{ __html: "You have Enrolled for this course"}} />
                        }
                        <div className="section section-b rel">                    
                            <h2 className="title s24 fontb">Course <span  className="fontn">Details</span></h2>
                            <div className="course-videos flex">
                                {courseVideos}
                            </div>
                        </div>
                    </div>
                </div>
            }
                <div className="course-preview rel video-align">
                    <div className="player rel">
                        {course && <video poster={course.poster} /> }
                        <div className="ctrls abs aic flex">
                            <button className="icon-pause s24 pp" />
                            <div className="timer rel fontn s15 cfff">
                                02:54 / 09:55
                            </div>
                            <div className="slider rel">
                                <div className="prog rel">
                                    <div className="bar rel">
                                        <div className="knob abs" />
                                    </div>
                                </div>
                            </div>
                            <button className="icon-volume-100 s24 vol" />
                            <button className="icon-full-screen-enter2 s24 fs" />
                        </div>
                    </div>
                </div>
            </div>
        )
    } catch (error) {
        return <HandleError error={error} />
    }
}

CourseDetails.propTypes = {
    match: PropTypes.object.isRequired,
    params: PropTypes.shape({
        courseid: PropTypes.number.isRequired
    })
}

export default CourseDetails;