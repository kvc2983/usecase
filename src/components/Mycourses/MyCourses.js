import React, {useEffect, useState} from "react";
import HandleError from '../Common/HandleError'
import {
    NavLink,    
} from "react-router-dom";

function MyCoursesPage(){

    const [course, setCourse] = useState(null);
    const [isPending, setIsPending] = useState(true);
    const [error, setError] = useState(null);
    const [deleteCourse, setDeleteCourse] = useState(true);

    useEffect(()=>{
        fetch('http://localhost:3000/mycourses')
        .then(res => {
            if(!res.ok) {
                throw Error("Endpoint for MyCourses is failed to fetch data");
            }
            return res.json();
        })
        .then(data => {
            setCourse(data);
            setIsPending(false);
            setError(null);
        })
        .catch(err => {
            console.log(err.message);
            setError(err.message);
            setIsPending(false);
        });
    }, [])

    const handleRemove = (id) => {
        fetch('http://localhost:3000/mycourses/' + id, {
            method: 'DELETE'
        });
        setDeleteCourse(false);
        window.location.reload();
    }

    var courseList = [];
    { isPending && <div>Loading...</div> }
    if(course) {
    for(let i = 0; i < course.length; i++){
        courseList.push(
            <div className="course-page rel flex">
                { error && <HandleError error={error} /> }
                <div className="course-info rel">

                <div className="tutor rel aic flex tutor">
                    <div className="pic">
                        <img src={course[i].tutor.dp} className="bl" alt="dp" />
                    </div>
                    <div className="meta rel">
                        <h2 className="s15 name fontb c333">{course[i].tutor.name}</h2>
                        <h2 className="s13 uname fontn c777">Course Tutor</h2>
                    </div>
                </div>

                
                <div >
                    <h2 className="s24 title fontb c333">{course[i].title}</h2>
                    <p className="s18 about fontn c777" dangerouslySetInnerHTML={{ __html: course[i].about}} />

                    {deleteCourse ?
                        <input type="button" id="Remove" className="lbl s20" onClick={(e) => (e.preventDefault(), handleRemove(course[i].id)) } value="Remove" /> 
                        :
                        <p  id="s18" className="s18 about fontn c777" dangerouslySetInnerHTML={{ __html: "Course removed from my courses"}} />
                    }

                    <div className="section section-b rel">
                        <h2 className="title s24 fontb">Course <span  className="fontn">Achievements</span></h2>
                        <div className="course-stats aic flex">
                            <div className="stats-box flex">
                                <div className="ico ico-points s24 icon-shield" />
                                <h2 className="val s15 c333 fontb">{course[i].points}</h2>
                                <h2 className="lbl s13 c777">points</h2>
                            </div>
                            <div className="stats-box flex">
                                <div className="ico ico-battery s24 icon-battery-90" />
                                <h2 className="val s15 c333 fontb">{course[i].percent}%</h2>
                                <h2 className="lbl s13 c777">complete</h2>
                            </div>
                            <div className="stats-box flex">
                                <div className="ico ico-battery s24 icon-battery-90" />
                                <h2 className="val s15 c333 fontb">{course[i].level}</h2>
                                <h2 className="lbl s13 c777">level</h2>
                            </div>
                        </div>
                    </div>
                </div>

                </div>
                <NavLink to={"/course/" + course[i].id} key={"popular-course-" + i}>
                    <div className="course-preview rel">
                        <div className="player rel video">
                            <video poster={course[i].poster} />
                            <div className="ctrls abs aic flex">
                                <button className="icon-pause s24 pp" />
                                <div className="timer rel fontn s15 cfff">
                                    02:54 / 09:55
                                </div>
                                <div className="slider rel">
                                    <div className="prog rel">
                                        <div className="bar rel">
                                            <div className="knob abs" />
                                        </div>
                                    </div>
                                </div>
                                <button className="icon-volume-100 s24 vol" />
                                <button className="icon-full-screen-enter2 s24 fs" />
                            </div>
                        </div>
                    </div>
                </NavLink>
            
            </div>
        );
    }
}
    
    try {
        return (
            <div className="app-page rel">
                {error &&
                    <div className="section rel">
                        <h2 className="title s24 fontb"><span  className="fontn">Something went wrong!</span></h2>
                        <div className="tutors rel flex">
                            <><small style={{ color: 'red' }}>{error}</small><br /></>
                        </div>
                    </div>
                }

                <h1 className="page-title s24 fontb c333">My Courses</h1>
                { course &&
                    <div className="section section-b rel">
                        <h2 className="title s24 fontb fontenroll"> Total <span  className="fontn">Enrolled Cources: </span>{course.length}</h2>
                    </div> }

                {  courseList }
            </div>
        )
    } catch (error) {
        return <HandleError error={error} />
    }

}

export default MyCoursesPage;