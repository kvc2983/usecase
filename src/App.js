import React, { useState, useEffect } from "react";
import './css/uifont.css';
import './css/App.css';
import './css/props.css';
//Module specific css files
import './css/CourseDetails.css';
import './css/Home.css';
import './css/SideBar.css';
import './css/RightBar.css';
import axios from 'axios';

//Left and Right Sidebar
import SideBar from "./SideBar/SideBar";
import RightBar from "./SideBar/RightBar";
//common Utils
import PrivateRoute from "./Utils/PrivateRoute";
import PublicRoute from "./Utils/PublicRoute";
import { getToken, removeUserSession, setUserSession } from "./Utils/Common";
import CourseDetails from "./components/Common/CourseDetails";
//Component Pages
import HomePage from "./components/Home/Home";
import AllCourses from "./components/AllCourses/AllCourses";
import MyCourses from "./components/Mycourses/MyCourses";
import WishList from "./components/Wishlist/Wishlist";
//Login
import Login from "./components/Login/Login";

import {
  Route,
  HashRouter
} from "react-router-dom";


function App() {
//Check User authentication also show Loading till User gets authenticated
  const [authLoading, setAuthLoading] = useState(true);

  useEffect(() => {
    const token = getToken();
    if (!token) {
      return;
    }

//Check User authentication and set loading to false on success
    axios.get(`http://localhost:4000/verifyToken?token=${token}`).then(response => {
      setUserSession(response.data.token, response.data.user);
      setAuthLoading(false);
    }).catch(error => {
      removeUserSession();
      setAuthLoading(false);
    });
  }, []);

  //Loading message while checking authentication
  if (authLoading && getToken()) {
    return <div className="content">Checking Authentication...</div>
  }

  return (
    <div className="App flex">
      <HashRouter>
        <SideBar />
          <div className="app-cotent">
            {/* PublicRoute URL is available without Login but for PrivateRout User needs to login First  */}
            <PrivateRoute path="/home" component={HomePage} />
            <Route exact path="/" component={Login} />
            <PublicRoute path="/login" component={Login} />
            <PrivateRoute path="/course/:courseid" component={CourseDetails} />
            <PrivateRoute path="/allcourses" component={AllCourses} />
            <PrivateRoute path="/my-courses" component={MyCourses} />
            <PrivateRoute path="/wishlist" component={WishList} />
          </div>
      </HashRouter>
      <RightBar />
    </div>
  );
}

export default App;
