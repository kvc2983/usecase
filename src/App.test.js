
import React from 'react';
import App from './App';
import SideBar from "./SideBar/SideBar";
import RightBar from "./SideBar/RightBar";
import HandleError from "./components/Common/HandleError";
import { shallow, mount } from 'enzyme';
import toJson from "enzyme-to-json";
import Rightbar from './SideBar/RightBar';
import CourseDetails from "./components/Common/CourseDetails";
import MyCourses from "./components/Mycourses/MyCourses";
import { render, screen, act } from "@testing-library/react";


describe("rendering components", () => {
  it("renders App component without crashing", () => {
    shallow(<App />);
  });
  
  it("renders Sidebar without crashing", () => {
    shallow(<SideBar />);
  });

  it("renders RightBar without crashing", () => {
    shallow(<RightBar />);
  });

  it("renders HandleError without crashing", () => {
    shallow(<HandleError />);
  });
});

//If In case there will be some update in the UI, then snapshot test will fail, so Need to update the snapshots.
describe("snapshots", () => {
  it("App snapshot for App component", () => {
    const tree = shallow(<App/>);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("App snapshot for SideBar component", () => {
    const sideBar = shallow(<SideBar/>);
    expect(toJson(sideBar)).toMatchSnapshot();
  });
  it("App snapshot for Rightbar component", () => {
    const rightBar = shallow(<Rightbar/>);
    expect(toJson(rightBar)).toMatchSnapshot();
  });
  it("App snapshot for HandleError component", () => {
    const handleError = shallow(<HandleError/>);
    expect(toJson(handleError)).toMatchSnapshot();
  });
});


describe("logic", () => {
  const match = { params : { courseid: 1 } };
  const wrapper = mount(<CourseDetails match={match} />);
  const course = mount(<MyCourses />);
  wrapper.find(".icon-pause").simulate("click");
  it("button click - Enroll Course", () => {
    // console.log(wrapper.debug());
    const timerVal = wrapper.find(".timer").text();
    const expectedValue = '02:54 / 09:55';
    expect(timerVal).toEqual(expectedValue);
  });
  it("Mycourses Page - title", () => {
    const removedValue = course.find(".page-title").text();
    const expectedBalanceValue = 'My Courses';
    expect(removedValue).toEqual(expectedBalanceValue);
  });
});

describe("Component", () => {
  it("Loads MyCourses Video Details", async () => {
    await act(async () => render(<MyCourses />));
    expect(screen.getByText("My Courses")).toBeInTheDocument;
  });
});
