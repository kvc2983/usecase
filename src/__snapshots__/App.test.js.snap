// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App snapshot for App component 1`] = `
<div
  className="App flex"
>
  <HashRouter>
    <Sidebar />
    <div
      className="app-cotent"
    >
      <PrivateRoute
        component={[Function]}
        path="/home"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <PublicRoute
        component={[Function]}
        path="/login"
      />
      <PrivateRoute
        component={[Function]}
        path="/course/:courseid"
      />
      <PrivateRoute
        component={[Function]}
        path="/allcourses"
      />
      <PrivateRoute
        component={[Function]}
        path="/my-courses"
      />
      <PrivateRoute
        component={[Function]}
        path="/wishlist"
      />
    </div>
  </HashRouter>
  <Rightbar />
</div>
`;

exports[`snapshots App snapshot for HandleError component 1`] = `
<div>
  <p>
    OOPS! Something went Wrong:
  </p>
  <p
    style={
      Object {
        "color": "red",
      }
    }
  />
  <p>
    Please try again after sometime
  </p>
</div>
`;

exports[`snapshots App snapshot for Rightbar component 1`] = `
<div
  className="rightbar rel"
>
  <div
    className="section rel"
  >
    <h2
      className="title s24 fontb"
    >
      Advanced 
      <span
        className="fontn"
      >
        Search
      </span>
    </h2>
    <div
      className="search-box rel flex"
    >
      <input
        className="qry s15 fontb"
        placeholder="Start writing something..."
        type="text"
      />
      <button
        className="go cfff s15 fontb"
      >
        Find
      </button>
    </div>
  </div>
  <div
    className="section section-b rel"
  >
    <h2
      className="title s24 fontb"
    >
      Popular 
      <span
        className="fontn"
      >
        This Week
      </span>
    </h2>
    <div
      className="courses rel flex"
    >
      <div
        className="course-a"
      />
      <div
        className="flex cols"
      />
    </div>
  </div>
  <div
    className="section section-b rel"
  >
    <h2
      className="title s24 fontb"
    >
      Special 
      <span
        className="fontn"
      >
        Offers
      </span>
    </h2>
    <div
      className="special-offer rel flex"
    >
      <div
        className="meta rel"
      >
        <h2
          className="h1 s20 fontb"
        >
          50 Days Premium!
        </h2>
        <h2
          className="h2 s14 fontb"
        >
          Get it before 5.5.2020
        </h2>
        <button
          className="s15 fontb"
        >
          Learn more
        </button>
      </div>
      <div
        className="vector rel"
      >
        <img
          alt="course"
          src="cake.png"
        />
      </div>
      <div
        className="ring abs"
      />
    </div>
  </div>
</div>
`;

exports[`snapshots App snapshot for SideBar component 1`] = `
<div
  className="sidebar rel"
>
  <a
    className="logo bl"
    href="#/home"
  >
    <img
      alt="Logo"
      className="bl"
      src="logo-coral.svg"
    />
  </a>
  <ul
    className="nav"
  >
    <li
      key="nav-0-home"
    >
      <a
        className="aic link noul flex c333"
        href="#/home"
      >
        <div
          className="ico s20 icon-home"
        />
        <h2
          className="lbl s20"
        >
          Home
        </h2>
      </a>
    </li>
    <li
      key="nav-1-allcourses"
    >
      <a
        className="aic link noul flex c333"
        href="#/allcourses"
      >
        <div
          className="ico s20 icon-ul"
        />
        <h2
          className="lbl s20"
        >
          All Courses
        </h2>
      </a>
    </li>
    <li
      key="nav-2-my-courses"
    >
      <a
        className="aic link noul flex c333"
        href="#/my-courses"
      >
        <div
          className="ico s20 icon-briefcase"
        />
        <h2
          className="lbl s20"
        >
          My Courses
        </h2>
      </a>
    </li>
    <li
      key="nav-3-wishlist"
    >
      <a
        className="aic link noul flex c333"
        href="#/wishlist"
      >
        <div
          className="ico s20 icon-tag"
        />
        <h2
          className="lbl s20"
        >
          Wishlist
        </h2>
      </a>
    </li>
  </ul>
  <div
    className="updated-course flex aic"
  >
    <div
      className="icon-lamp-bright cfff s24 ico"
    />
    <div
      className="lbl s15 fontb c333"
    >
      Updated Courses
      <h2
        className="author s13 c777"
      >
        by Kunal Chandegaonkar
      </h2>
    </div>
  </div>
  <div
    className="stats aic flex"
  >
    <div
      className="stats-box flex"
    >
      <div
        className="ico ico-points s24 icon-shield"
      />
      <h2
        className="val s15 c333 fontb"
      >
        1800
      </h2>
      <h2
        className="lbl s13 c777"
      >
        points
      </h2>
    </div>
    <div
      className="stats-box flex"
    >
      <div
        className="ico ico-battery s24 icon-battery-90"
      />
      <h2
        className="val s15 c333 fontb"
      >
        45.3%
      </h2>
      <h2
        className="lbl s13 c777"
      >
        complete
      </h2>
    </div>
  </div>
</div>
`;
